src/sagemaker/utils.py

```diff
@@ -649,4 +651,61 @@ def _module_import_error(py_module, feature, extras):
     return error_msg.format(py_module, feature, extras)
 
 
+class NotebooksDataConfig(abc.ABC):
+    """Abstract base class for creation for accessing data related to Example Notebooks.
+
+    Provides a skeleton for customization by overriding of method fetch_data_config.
+    """
+
+    @abc.abstractmethod
+    def fetch_data_config(self, sagemaker_session):
+        """Abstract method implementing retrieval of data config from a pre-configured data source.
+
+        Args:
+            sagemaker_session (Session): SageMaker session instance to use for boto configuration
+
+        Returns:
+            object: The data configuration object for Example Notebooks
+        """
+
+
+class S3DataConfig(NotebooksDataConfig):
+    """This class extends the NotebooksDataConfig class to fetch a data config file hosted on S3"""
+
+    @staticmethod
+    def fetch_data_config(sagemaker_session):
+        """Fetches data configuration for Example Notebooks from a S3 bucket.
+
+        Args:
+            sagemaker_session (Session): SageMaker session instance to use for boto configuration.
+
+        Returns:
+            object: The JSON object containing data configuration.
+        """
+
+        bucket_name = "example-notebooks-data-config"
+        prefix = "config/data_config.json"
+        json_string = sagemaker_session.read_s3_file(bucket_name, prefix)
+        return json.loads(json_string)
+
+    @staticmethod
+    def get_data_bucket(sagemaker_session):
+        """Returns the bucket containing the data for specified region.
+
+        Args:
+            sagemaker_session (Session): SageMaker session instance to use for boto configuration.
+
+        Returns:
+            str: The S3 bucket containing datasets for Example Notebooks in the specified region.
+        """
```

Comment:
This returns config and not bucket name right

Reply:
This returns specifically a bucket and is a method specific to this subclass.
The data config is fetched from another method, S3DataConfig.fetch_data_config which this uses